bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -188,32 +220,40 @@ public BigtableBufferedMutator(
     this.asyncExecutor = session.createAsyncExecutor();
     this.bulkOptions = options.getBulkOptions();
     this.executorService = asyncRpcExecutorService;
+    int asyncMutatorCount = options.getBulkOptions().getAsyncMutatorCount();
+    this.asyncOperationsQueue =
+        asyncMutatorCount == 0 ? null : new LinkedBlockingQueue<Runnable>(asyncMutatorCount * 4);
     if (bulkOptions.useBulkApi()) {
       BigtableTableName tableName = this.adapter.getBigtableTableName();
       this.bulkMutation = session.createBulkMutation(tableName, asyncExecutor);
     }
   }
 
   private void initializeAsyncMutators() {
-    if (executorService != null && activeMutationWorkers.get() < bulkOptions.getAsyncMutatorCount()) {
+    if (areNewWorkersRequired()) {
       synchronized (activeMutationWorkers) {
-        for (int i = activeMutationWorkers.get(); i < bulkOptions.getAsyncMutatorCount(); i++) {
-          executorService.submit(mutationWorker);
+        if (areNewWorkersRequired()) {
+          for (int i = activeMutationWorkers.get(); i < bulkOptions.getAsyncMutatorCount(); i++) {
+            executorService.submit(mutationWorker);
+          }
         }
       }
     }
   }
 
+  protected boolean areNewWorkersRequired() {
+    return executorService != null
+        && (activeMutationWorkers.get() < bulkOptions.getAsyncMutatorCount()
+            || lastOperationFoundInMs < System.currentTimeMillis() - ASYNC_POLL_WAIT_MS);
```

Comment:
I'm not quite sure why lastOperationFoundInMs is needed, I'm probably just missing it. Comment might be helpful. Also, I think it would need to be volatile to make sure the latest value, published in a worker thread, is visible here in the main thread.